src/threshold.c

```diff
@@ -915,7 +916,8 @@ int ut_config (oconfig_item_t *ci)
       break;
   }
 
-  if (c_avl_size (threshold_tree) > 0) {
+  /* register callbacks if this is the first time we see a valid config */
+  if (old_size == 0 && c_avl_size (threshold_tree) > 0) {
```

Comment:
Nit: Common collectd style adds parenthesis around each case in an if statement:

    if ((old_size == 0) && ...)